UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetByProductVersion.cs

```diff
@@ -0,0 +1,148 @@
+﻿using NUnit.Framework;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+using UKHO.ExchangeSetService.API.FunctionalTests.Helper;
+using UKHO.ExchangeSetService.API.FunctionalTests.Models;
+
+namespace UKHO.ExchangeSetService.API.FunctionalTests.FunctionalTests
+{
+    class ExchangeSetByProductVersion
+    {
+        private ExchangeSetApiClient ExchangesetApiClient { get; set; }
+        private TestConfiguration Config { get; set; }
+        public DataHelper Datahelper { get; set; }
+        public ProductVersionModel ProductVersionmodel { get; set; }
+      
+        [SetUp]
+        public void Setup()
+        {
+            Config = new TestConfiguration();
+            ExchangesetApiClient = new ExchangeSetApiClient(Config.EssBaseAddress);
+            ProductVersionmodel = new ProductVersionModel();
+            Datahelper = new DataHelper();
+           
+        }
+
+        [Test]
+        public async Task WhenICallTheApiWithAValidProductVersion_ThenASuccessStatusIsReturned()
```

Comment:
This is verifying far more than just the status.. maybe _ThenTheCorrectResponseIsReturned ?

Reply:
Updated the Test Name to "WhenICallTheApiWithAValidProductVersion_ThenTheCorrectResponseIsReturned()".